load/dependencyresolver.go

```diff
@@ -174,3 +193,157 @@ func getNearestAncestor(g *graph.Graph, id, node string) (string, bool) {
 	}
 	return siblingID, true
 }
+
+func withoutRoot(in []string) (out []string) {
+	for _, id := range in {
+		if !graph.IsRoot(id) {
+			out = append(out, id)
+		}
+	}
+	return out
+}
+
+func withoutModule(g *graph.Graph, in []string) (out []string) {
+	for _, id := range in {
+		if meta, ok := g.Get(id); ok {
+			if node, ok := meta.Value().(*parse.Node); ok {
+				if !node.IsModule() {
+					out = append(out, id)
+				}
+			}
+		}
+	}
+	return out
+}
+
+func withoutSelf(self string, in []string) (out []string) {
+	for _, id := range in {
+		if id != self {
+			out = append(out, id)
+		}
+	}
+	return out
+}
+
+func highestEdge(g *graph.Graph, id string) string {
+	edges := graph.Sources(g.UpEdges(id))
+	for _, edge := range edges {
+		if !graph.IsRoot(edge) && edge != id {
+			return highestEdge(g, edge)
+		}
+	}
+	return id
+}
+
+type byDependencyCount struct {
+	g     *graph.Graph
+	nodes []*node.Node
+}
+
+func (b byDependencyCount) Len() int      { return len(b.nodes) }
+func (b byDependencyCount) Swap(i, j int) { b.nodes[i], b.nodes[j] = b.nodes[j], b.nodes[i] }
+func (b byDependencyCount) Less(i, j int) bool {
+	return len(b.g.Dependencies(b.nodes[i].ID)) > len(b.g.Dependencies(b.nodes[j].ID))
+}
+
+func groupDeps(ctx context.Context, g *graph.Graph, group string) (*graph.Graph, error) {
```

Comment:
Is there any way that we can reasonably break this up to factor out smaller helper functions?  Like maybe pulling some of the individual for loops out into helper functions?